lib/api/pokedex_api_spec/handler/pokemon_handler.dart

```diff
@@ -22,4 +24,14 @@ class PokemonApi {
       return response.data[str.results].map<Pokemon>((dynamic data) => Pokemon.fromJson(data as Json)).toList();
     });
   }
+
+  Future<PokemonDto> getPokemon(String id) async {
+    final queryParams = <String, dynamic>{};
+
+    final baseUri = Uri.parse(apiClient.dio.options.baseUrl);
+    final uri = baseUri.replace(queryParameters: queryParams, path: '${baseUri.path}/pokemon/$id');
+    return await apiClient.dio.getUri(uri).then((response) {
+      return Pokemon.fromJson(response.data).toDto(int.parse(id));
```

Comment:
@yohancai `handler` is supposed to be the bridge in communicating with the api and supposed to be returning the raw result it receives. For your desire of mapping it, I would suggest do it in the respective `action` 🙂